projects/valtimo/zgw/src/lib/modules/zaakdetails/components/document-objecten-api-sync/document-objecten-api-sync.component.ts

```diff
@@ -0,0 +1,195 @@
+    return this.formGroup.get('objectManagementConfigurationId');
+  }
+
+  public get enabled(): AbstractControl<boolean> {
+    return this.formGroup.get('enabled');
+  }
+
+  public readonly valid$ = new BehaviorSubject<boolean>(false);
+  private _subscriptions = new Subscription();
+
+  constructor(
+    private readonly route: ActivatedRoute,
+    private readonly documentObjectenApiSyncService: DocumentObjectenApiSyncService
+  ) {}
+
+  public ngOnInit(): void {
+    this.loadDocumentenObjectenApiSync();
+  }
+
+  public ngOnDestroy(): void {
+    this._subscriptions.unsubscribe();
+  }
+
+  public loadDocumentenObjectenApiSync(): void {
+    this._subscriptions.add(
+      this.documentDefinition$
+        .pipe(
+          switchMap(documentDefinition =>
+            this.documentObjectenApiSyncService.getDocumentObjectenApiSync(
+              documentDefinition.id.name,
+              documentDefinition.id.version
+            )
+          )
+        )
+        .subscribe(documentObjectenApiSync => {
+          this.loading$.next(false);
+          this.configSelected(documentObjectenApiSync?.objectManagementConfigurationId);
+          this.enabled.patchValue(documentObjectenApiSync?.enabled);
+          this.documentObjectenApiSync$.next(documentObjectenApiSync);
+        })
+    );
+  }
+
+  public remove(): void {
+    this._subscriptions.add(
+      this.documentDefinition$
+        .pipe(
+          switchMap(documentDefinition =>
+            this.documentObjectenApiSyncService.deleteDocumentObjectenApiSync(
+              documentDefinition.id.name,
+              documentDefinition.id.version
+            )
+          ),
+          tap(() => {
+            this.documentObjectenApiSync$.next(null);
+          })
+        )
+        .subscribe()
+    );
+  }
+
+  public submit(): void {
+    const formValues = this.formGroup.getRawValue();
+    this.documentDefinition$
+      .pipe(
+        switchMap(documentDefinition =>
+          this.documentObjectenApiSyncService.updateDocumentObjectenApiSync(
+            documentDefinition.id.name,
+            documentDefinition.id.version,
+            {
+              objectManagementConfigurationId: formValues.objectManagementConfigurationId,
+              enabled: formValues.enabled,
+            }
+          )
+        )
+      )
+      .subscribe(() => {
+        this.loadDocumentenObjectenApiSync();
+        this.hideModal();
+      });
+  }
+
+  public onModalClose(): void {
+    this.hideModal();
+  }
+
+  public showModal(): void {
+    this.modalShowing$.next(true);
+  }
+
+  public configSelected(selectedId: string): void {
+    if (!selectedId) {
+      this.objectManagementConfigurationId.patchValue('');
+    } else {
+      this.objectManagementConfigurationId.patchValue(selectedId);
+    }
+  }
+
+  private hideModal(): void {
+    this.modalShowing$.next(false);
+  }
+
+  protected readonly CARBON_THEME = CARBON_THEME;
```

Comment:
I don't get this, CARBON_THEME is an enum but used as a value in template. Also, carbon theme should change depending on light or dark mode, can't hardcode it. Search for theme$ in code for example on how to respond to current theme changes